src/coap_client.c

```diff
@@ -486,6 +487,55 @@ enum golioth_status golioth_coap_client_observe_async(struct golioth_client *cli
     return GOLIOTH_OK;
 }
 
+enum golioth_status golioth_coap_client_observe_release_async(
+    struct golioth_client *client,
+    const char *path_prefix,
+    const char *path,
+    enum golioth_content_type content_type,
+    uint8_t *token,
+    size_t token_len,
+    void *arg)
+{
+    if (!client || !path)
+    {
+        return GOLIOTH_ERR_NULL;
+    }
+
+    if (!client->is_running)
+    {
+        GLTH_LOGW(TAG,
+                  "Client not running, dropping observe release request for path %s%s",
+                  path_prefix,
+                  path);
+        return GOLIOTH_ERR_INVALID_STATE;
+    }
+
+    golioth_coap_request_msg_t request_msg = {
+        .type = GOLIOTH_COAP_REQUEST_OBSERVE_RELEASE,
+        .path_prefix = path_prefix,
+        .ageout_ms = GOLIOTH_SYS_WAIT_FOREVER,
+        .observe =
+            {
+                .content_type = content_type,
+                .arg = arg,
+            },
+    };
+    strncpy(request_msg.path, path, sizeof(request_msg.path) - 1);
```

Comment:
Would be good to check whether `path` fits into `request_msg.path`.

Reply:
Is there a good way to get the length of this? It's a `const char *path` being passed in and I don't think I can safely call str(len) on that.

Reply:
You should be able to use `strlen()` without any issues. If the length won't fit into `request_msg.path` then we should return some meaningful error code.